src/devices/video/upd7220.cpp

```diff
@@ -1323,6 +1327,12 @@ void upd7220_device::process_fifo()
 
 		if (m_param_ptr == 3 || (m_param_ptr == 2 && m_cr & 0x10))
 		{
+			if (m_figs.m_figure_type)
+			{
+				m_pattern = (m_pr[2] << 8) | m_pr[1];
```

Comment:
So setting the m_pattern probably can and probably should be moved outside this if, since even though the wdat function does not currently use it. The pattern register should be set with wdats paramters regardless, from what I can tell from the datasheet.

More importantly I think you are setting it wrong here. Here you are currently assuming that the wdat command type is set to word access, rather then byte access. you should check the type and if its uses word access do this, otherwise there will only be a single byte paramter which should either be set as the high or low byte of the pattern register with the other byte being set to zeros.